src/components/career-form/career-file-upload.tsx

```diff
@@ -0,0 +1,238 @@
+import { Input, InputContainer } from '../form/controls';
+import React, { ReactNode, useCallback } from 'react';
+import { DropzoneRootProps, useDropzone } from 'react-dropzone';
+import styled from 'styled-components';
+import { theme } from '../layout/theme';
+import {
+  FieldErrors,
+  FieldValues,
+  UseFormClearErrors,
+  UseFormRegister,
+  UseFormSetError,
+  UseFormSetValue,
+  UseFormWatch,
+} from 'react-hook-form';
+import { GridItem } from '../grid/grid';
+import { FilePreview } from './career-file-preview';
+import { FormError } from './career-components';
+import { CareerSelect } from './career-select';
+import { FileTitleWrapper } from './career-file-title';
+
+interface FileUploadProps {
+  setValue: UseFormSetValue<FieldValues>;
+  setError: UseFormSetError<FieldValues>;
+  clearErrors: UseFormClearErrors<FieldValues>;
+  register: UseFormRegister<FieldValues>;
+  watch: UseFormWatch<FieldValues>;
+  name: string;
+  selectedFiles: FileList;
+  errors: FieldErrors;
+  children: ReactNode | ReactNode[];
+}
+
+const MAX_NUMBER = 3;
+const MAX_SIZE = 20000000; // maximal size per document of personio
+
+export const FileUpload = ({
+  setValue,
+  setError,
+  clearErrors,
+  name,
+  register,
+  selectedFiles,
+  errors,
+  children,
+  watch,
+}: FileUploadProps) => {
+  const onDrop = useCallback(
+    (droppedFiles) => {
+      // maximal number of file being dropped
+      const maximalOfFiles = selectedFiles
+        ? MAX_NUMBER - selectedFiles.length
+        : MAX_NUMBER;
+      // dropped files are appended to already selected files
+      const dataTransfer = new DataTransfer();
+      if (selectedFiles) {
+        for (let i = 0; i < selectedFiles.length; i++) {
+          dataTransfer.items.add(selectedFiles[i]);
+        }
+      }
+      for (let i = 0; i < Math.min(maximalOfFiles, droppedFiles.length); i++) {
+        dataTransfer.items.add(droppedFiles[i]);
+      }
+      setValue(name, dataTransfer.files, { shouldDirty: true });
+      clearErrors(name);
+
+      // in case not all file could be added, an error is set
+      if (
+        Math.min(maximalOfFiles, droppedFiles.length) !== droppedFiles.length
+      ) {
+        setError(
+          name,
+          {
+            type: 'manual',
+            message: 'Maximal number of documents is reached.',
+          },
+          { shouldFocus: true },
+        );
+      }
+    },
+    [selectedFiles, setValue],
+  );
+
+  const validator = (file) => {
+    clearErrors(name);
+    if (selectedFiles) {
+      // check if file was already uploaded
+      for (let i = 0; i < selectedFiles.length; i++) {
+        if (JSON.stringify(selectedFiles[i]) === JSON.stringify(file)) {
```

Comment:
`JSON.stringify` can get pretty expansive  and hang peoples browser for large files, I would simply use the file name 😬 You never know what gigantic file people try to upload (although I would question if they.. well other topic)